services/tmc/src/pages/api/model-solution.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable i18next/no-literal-string */
+import axios from "axios"
+import { promises as fs } from "fs"
+import { NextApiRequest, NextApiResponse } from "next"
+import { temporaryDirectory, temporaryFile } from "tempy"
+
+import { ClientErrorResponse } from "../../lib"
+import { RepositoryExercise } from "../../shared-module/bindings"
+import { isRepositoryExercise } from "../../shared-module/bindings.guard"
+import { isString } from "../../shared-module/utils/fetching"
+import { compressProject, extractProject, prepareSolution } from "../../tmc/langs"
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+  if (req.method !== "POST") {
+    return res.status(404).json({ message: "Not found" })
+  }
+
+  return await handlePost(req, res)
+}
+
+type ModelSolutionRequest = RepositoryExercise & { upload_url: string }
+
+const handlePost = async (
+  req: NextApiRequest,
+  res: NextApiResponse<void | ClientErrorResponse>,
+): Promise<void> => {
+  let request: ModelSolutionRequest
+  const upload_url = req.body["upload_url"]
+  if (isRepositoryExercise(req.body) && isString(upload_url)) {
```

Comment:
Not entirely happy that this takes the repository exercise directly because
a) it exposes the columns in our database to the exercise services, making some of them depend on the table structure staying unchanged
b) These endpoints are supposed to take the private spec as an input and that may not be the same as the repository exercise

So maybe we should create an endpoint that generates the private spec from some simplified representation of the repository exercise and then we can use that private spec as an input to these endpoints.